connector-runtime/connector-runtime-core/src/test/java/io/camunda/connector/runtime/core/outbound/ConnectorJobHandlerTest.java

```diff
@@ -862,6 +862,27 @@ void shouldCreateBpmnError_UsingExceptionCodeAndErrorVariables() {
       assertThat(result.getErrorMessage()).isEqualTo("Message: exception message");
     }
 
+    @Test
+    void shouldHideSecretsInJobErrorMessage() {
+      // given
+      var errorMessage = "Something went wrong: bar is not the correct password";
+      var jobHandler =
+          newConnectorJobHandler(
+              context -> {
+                throw new IllegalArgumentException(errorMessage);
+              });
+
+      // when
+      var result =
+          JobBuilder.create()
+              .withVariables("{{secrets.FOO}}")
+              .executeAndCaptureResult(jobHandler, false);
+
+      // then
+      assertThat(result.getErrorMessage())
+          .isEqualTo("Something went wrong: *** is not the correct password");
+    }
+
```

Comment:
Please add additional tests that:
* Show an example where we provide a real json object  with the variables
* Show how a validation exception looks like when the value(coming from a secret) is not passing validation and shows up obfuscated in the validation message.